src/components/play/PlayView.tsx

```diff
@@ -68,21 +68,58 @@ const PlayView = () => {
   }, [newResponse]);
 
   const onSubmit = () => {
-    setShowCorrection(true);
-    if (newResponse.id) {
-      patchAppData({
-        id: newResponse.id,
-        data: newResponse.data,
-      });
+    if (newResponse) {
+      setShowCorrection(true);
+      if (newResponse.id) {
+        patchAppData({
+          id: newResponse.id,
+          data: newResponse.data,
+        });
+      } else {
+        if (newResponse.data) {
+          postAppData({
+            data: newResponse.data,
+            type: APP_DATA_TYPES.RESPONSE,
+          });
+        } else {
+          console.error('The response data is not defined !');
+        }
+      }
     } else {
-      postAppData({
-        data: newResponse.data,
-        type: APP_DATA_TYPES.RESPONSE,
-      });
+      console.error('The response is not defined !');
     }
   };
 
-  if (!questions || questions.isEmpty()) {
+  const setInData = <
+    T extends AppData,
+    K extends keyof T['data'],
+    V extends T['data'][K]
+  >(
+    object: Partial<T>,
+    key: K,
+    value: V
+  ): Partial<T> => {
+    return {
+      ...object,
+      data: {
+        ...object.data,
+        [key]: value,
+      },
+    };
+  };
+
+  const setIn = <T extends AppData, K extends keyof T, V extends T[K]>(
+    object: Partial<T>,
+    key: K,
+    value: V
+  ): Partial<T> => {
+    return {
+      ...object,
+      [key]: value,
+    };
+  };
```

Comment:
Would this make sense in a util file? I think you already defined a similar function in another file.

Reply:
I can move this in a util file. But the functions are not the same. One is to copy an array and update the copy, and the two others are use to copy an object and update it. So, I can't merge those functions together...